client/kubernetes/cluster.go

```diff
@@ -0,0 +1,132 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tinkerbell/tink/pkg/apis/core/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+)
+
+const (
+	WorkflowWorkerNonTerminalStateIndex = ".status.state.nonTerminalWorker"
+	HardwareMACAddrIndex                = ".spec.interfaces.dhcp.mac"
+	HardwareIPAddrIndex                 = ".spec.interfaces.dhcp.ip"
+)
+
+var (
+	runtimescheme = runtime.NewScheme()
+)
+
+func init() {
+	_ = clientgoscheme.AddToScheme(runtimescheme)
+	_ = v1alpha1.AddToScheme(runtimescheme)
+}
```

Comment:
we're trying to minimize `init()s` can this be done some other way? sync.Once.Do() in NewCluster maybe?

Reply:
Done